src/freertos_drivers/ti/TivaDCC.hxx

```diff
@@ -945,13 +945,13 @@ TivaDCC<HW>::TivaDCC(const char *name, RailcomDriver* railcom_driver)
 
     unsigned h_deadband = 2 * (HW::H_DEADBAND_DELAY_NSEC / 1000);
     unsigned railcom_part = 0;
-    fill_timing(RAILCOM_CUTOUT_PRE, 6 - railcom_part, 0);
```

Comment:
The code here has an invariant that railcom_part is always set to the absolute microsecond that the sum of the previous RAILCOM_CUTOUT_XXX segments represent. This is used to compute the length of next part by a simple subtraction.
You broke this invariant, and after your changes it is impossible to reason what this code does. Please restore this invariant so we can merge this code.